integrations/tensorflow/e2e/keras/iree_vision_test_suite.bzl

```diff
@@ -15,35 +15,107 @@
+            name,
+            model,
+            dataset,
+            reference_backend,
+            backend,
+        )
+        tests.append(test_name)
+
+        args = [
+            "--model={}".format(model),
+            "--data={}".format(dataset),
+            "--include_top=1",
+            "--target_backends={},{}".format(reference_backend, backend),
+        ]
+        if external_weights:
+            args.append("--url={}".format(external_weights))
+
+        # TODO(GH-2175): Simplify this after backend names are standardized.
+        driver = backend.replace("iree_", "")  # "iree_<driver>" --> "<driver>"
+        if driver == "llvmjit":
+            driver = "llvm"
+        py_test_tags = ["driver={}".format(driver)]
+        if tags != None:  # `is` is not supported.
+            py_test_tags += tags
+
+        iree_py_test(
+            name = test_name,
+            main = "vision_model_test.py",
+            srcs = ["vision_model_test.py"],
+            args = args,
+            tags = py_test_tags,
+            deps = deps,
+            size = size,
+            python_version = python_version,
+            **kwargs
+        )
+
+    native.test_suite(
+        name = name,
+        tests = tests,
+        # Note that only the manual tag really has any effect here. Others are
+        # used for test suite filtering, but all tests are passed the same tags.
+        tags = tags,
+        # If there are kwargs that need to be passed here which only apply to
+        # the generated tests and not to test_suite, they should be extracted
+        # into separate named arguments.
+        **kwargs
+    )
+
+def iree_vision_test_suite(
+        name,
+        models,
         datasets,
-        models_to_backends,
+        backends,
+        reference_backend,
+        failing_configurations = None,
         external_weights = None,
         tags = None,
         deps = None,
         size = "large",
         python_version = "PY3",
         **kwargs):
-    """Expands a set of iree_py_tests for the specified vision models and bundles them into a test suite.
+    """Expands two sets of iree_py_tests from the specified models, datasets, backends and failing configurations and bundles them into a test suite.
 
     Creates one test per dataset, backend, and model.
 
     Args:
       name:
-        name of the generated test suite.
+        name of the generated passing test suite. If failing_configurations is
+        not `None` then a test suite named name_failing will also be generated.
+      models:
+        an iterable of model names to generate targets for.
+      datasets:
+        a iterable specifying the dataset on which the model is based. This
+        controls the shape of the input images. Also indicates which weight file
+        to use when loading weights from an external source.
+      backends:
+        an iterable of targets backends to generate targets for, for each model
+        and for each dataset.
```

Comment:
Why is this one suffixed with "for each model and for each dataset", but the others aren't? I'd just leave this off for all and note that you're doing the full expansion